mythril/laser/ethereum/state/memory.py

```diff
@@ -0,0 +1,99 @@
+from typing import Union
+from z3 import BitVecRef, Extract, BitVecVal, If, BoolRef, Z3Exception, simplify, Concat
+
+from mythril.laser.ethereum import util
+
+
+class Memory:
+    def __init__(self):
+        self._memory = []
+
+    def __len__(self):
+        return len(self._memory)
+
+    def extend(self, size):
+        self._memory.extend(bytearray(size))
+
+    def get_word_at(self, index: int) -> Union[int, BitVecRef]:
+        """
+        Access a word from a specified memory index
+        :param index: integer representing the index to access
+        :return: 32 byte word at the specified index
+        """
+        try:
+            return util.concrete_int_from_bytes(
+                bytes([util.get_concrete_int(b) for b in self[index : index + 32]]), 0
+            )
+        except TypeError:
+            return simplify(
+                Concat(
+                    [
+                        b if isinstance(b, BitVecRef) else BitVecVal(b, 256)
+                        for b in self[index : index + 32]
+                    ]
+                )
+            )
+
+    def write_word_at(self, index: int, value: Union[int, BitVecRef, bool, BoolRef]):
+        """
+        Writes a 32 byte word to memory at the specified index`
+        :param index: index to write to
+        :param value: the value to write to memory
+        """
+        try:
+            # Attempt to concretize value
+            if isinstance(value, bool):
+                _bytes = (
+                    int(1).to_bytes(32, byteorder="big")
+                    if value
+                    else int(0).to_bytes(32, byteorder="big")
+                )
+            _bytes = util.concrete_int_to_bytes(value)
```

Comment:
Shouldn't this be in an `else`? As `value=True/False` might raise an AttributeError here.